producer-c/producer-cloudwatch-integ/CanaryStreamUtils.cpp

```diff
@@ -0,0 +1,228 @@
+/**
+ * Kinesis Video Producer Continuous Retry Stream Callbacks
+ */
+#define LOG_CLASS "CanaryStreamCallbacks"
+#include "CanaryStreamUtils.h"
+
+STATUS createCanaryStreamCallbacks(Aws::CloudWatch::CloudWatchClient* cwClient,
+                                   PCHAR pStreamName,
+                                   PStreamCallbacks* ppStreamCallbacks, PCanaryStreamCallbacks* ppCanaryStreamCallbacks)
```

Comment:
You don't need to return a PStreamCallbacks pointer which is cast pointer to PCanaryStreamCallbacks. This is a specialized function that returns the CanaryStreamCallbacks - caller should have this knowledge. The caller should then cast it to StreamCallbacks.
This is the pattern we use elsewhere - for example the CurlCallbacks, etc..

Reply:
Agreed! The initial design was to pass around streamCallbacks. Somehow, it ended up staying. Fixed!